device/core/src/module_client.ts

```diff
@@ -83,8 +92,7 @@ export class ModuleClient extends InternalClient {
         debug('transport disconnect event:  no error');
       }
       if (err && this._retryPolicy.shouldRetry(err)) {
-        if (this._inputMessagesEnabled) {
-          this._inputMessagesEnabled = false;
+        if (this._userRegisteredInputMessageListener) {
```

Comment:
We want to re-enable input messages if the user registered a listener.
Before, we were enabling if the user registered a listener AND the transport successfully enabled input messages prior.